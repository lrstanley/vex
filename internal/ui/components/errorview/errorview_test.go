// Copyright (c) Liam Stanley <liam@liam.sh>. All rights reserved. Use of
// this source code is governed by the MIT license that can be found in
// the LICENSE file.

package errorview

import (
	"errors"
	"strings"
	"testing"

	"github.com/lrstanley/vex/internal/ui/testui"
)

func TestMain(m *testing.M) {
	testui.TestMain(m)
}

func TestNew(t *testing.T) {
	t.Run("2 errors", func(t *testing.T) {
		m := New()
		m.SetHeight(testui.DefaultTermHeight)
		m.SetWidth(testui.DefaultTermWidth)
		tm := testui.NewNonRootModel(t, m, false)
		m.SetErrors(
			errors.New("test error 1"),
			errors.New("test error 2"),
		)
		tm.ExpectViewContains(t, "2 errors", "test error 1", "test error 2")
		tm.ExpectViewDimensions(t, m.GetWidth(), m.GetHeight())
		tm.ExpectViewSnapshot(t)
	})

	t.Run("too-many-errors", func(t *testing.T) {
		m := New()
		m.SetHeight(testui.DefaultTermHeight)
		m.SetWidth(testui.DefaultTermWidth)
		tm := testui.NewNonRootModel(t, m, false)
		m.SetErrors(
			errors.New(strings.Repeat("test error\n", 100)),
		)
		tm.ExpectViewContains(t, "1 errors", "error(s) not shown")
		tm.ExpectViewDimensions(t, m.GetWidth(), m.GetHeight())
		tm.ExpectViewSnapshot(t)
	})

	t.Run("0-width-height", func(t *testing.T) {
		m := New()
		m.SetHeight(0)
		m.SetWidth(0)
		tm := testui.NewNonRootModel(t, m, false)
		m.SetErrors(errors.New("test error"))
		tm.ExpectViewSnapshot(t)
	})
}
