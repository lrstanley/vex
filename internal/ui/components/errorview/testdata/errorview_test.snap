
[TestNew/2_errors - 1]
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                     [m[m2 errors[m [m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m                      
                    [m [m[mâ€¼ test error 1[m[m [m[m                       [m                     
                                                                                
                    [m [m[mâ€¼ test error 2[m[m [m[m                       [m                     
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
---

[TestNew/too-many-errors - 1]
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                     [m[m1 errors[m [m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m/[m[m                      
                    [m [m[mâ€¼ 1 error(s) not shown (screen size)[m[m [m[m [m                     
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
---

[TestNew/0-width-height - 1]

---
